ludwig/features/binary_feature.py

```diff
@@ -167,10 +167,9 @@ def predictions(
             self,
             inputs  # hidden
     ):
-        logits, _ = self.logits(input)
 
         probabilities = tf.nn.sigmoid(
-            logits,
+            inputs,
```

Comment:
this should probably be logits not inputs. I expect input to be [batch, hidden], and logits to be [batch, 1] or [batch,] (not sure, dpendneds on the dimensions of the y, most likely the second is correct)

Reply:
OK...I understand your rationale.  The fix for this comment is to add this statement `logits = inputs` in the following code fragment
```
    def predictions(
            self,
            inputs  # hidden
    ):
        logits = inputs
        probabilities = tf.nn.sigmoid(
            logits,
            name='probabilities_{}'.format(
                self.name)
        )
```
During debugging, I noticed that `inputs` parameter has shape (batch,).  So if `inputs` should be a matrix, e.g., shape (batch, hidden), then I'll need to review how the different layers are computed for the binary feature.  

In the case of NumericOutputFeature, `inputs` is a matrix of shape (batch, hidden).

After I take another look,  I may be back with some questions on this.

Reply:
Yeah it should be a matrix.. An there should be a function that calculates the logit exactly identical to numerical, then probabilities is obtained by `sigmoid` and predictions are obtained by `probablities > threshold`

Roughly:
```
    def predictions(
            self,
            inputs,  # logits
    ):
        logits = inputs
        probabilities = tf.nn.sigmoid(
            logits,
            name='probabilities_{}'.format(
                self.name)
        )
        predictions = probablities > self.threshold
        return {
            self.feature_name: {'predictions': predictions, 'probablities': probablities, 'logits': logits}
        }
```

Reply:
From this exchange https://github.com/uber/ludwig/pull/646#issuecomment-603579026 (last part of posting) and https://github.com/uber/ludwig/pull/646#issuecomment-603581906  I believe the fix for this comment is to revise the method signature for the <Type>OutputFeature.predictions()` from this (highlighted code)
```
    def predictions(
            self,
            inputs                                    #<====change
    ):

        probabilities = tf.nn.sigmoid(
            inputs,                                     #<=======change
            name='probabilities_{}'.format(
                self.name)
        )
```
to this
```
    def predictions(
            self,
            logits                                 #<=======to this
    ):

        probabilities = tf.nn.sigmoid(
            logits,                               #<=========to this
            name='probabilities_{}'.format(
                self.name)
        )
```

For the future work, I need to keep in mind the expected shape for `logits` is (batch,) and `inputs` is a matrix with shape (batch, hidden)

If this right, I'll make the change for the Numerical and Binary features.

Reply:
Or maybe another way to ask...from a variable naming standards, should I ensure the following:
* a variable called `logits` is a tensor with rank 1
* a variable called `inputs` is a tensor with rank 2 or greater (I'm assuming greater when we get to images and sequences).

Reply:
i would say `logits()` maps hidden representations to predictions pre-softmax or pre-sigmoid. `predictions()` uses the output of logits to calculate the probabilities and the argmax (in case of categorical features). Usually the inputs to `logits()` are one rank mode, for instance for binary they are [b,h] and ` logits()` output is `[b,1]`, then predictions take this `[b,1]` and passes through a sigmoid, which leaves it `[b,1]`. Does this make sense?

Reply:
Yes it makes sense.I just need more exposure to how different data types are handled.  I implemented the guidance from here https://github.com/uber/ludwig/pull/646#discussion_r398300024